packages/core/src/dialog/DialogContent.tsx

```diff
@@ -24,11 +31,15 @@ export interface DialogContentProps extends HTMLAttributes<HTMLDivElement> {
 export const DialogContent = forwardRef<HTMLDivElement, DialogContentProps>(
   function DialogContent(props, ref) {
     const { children, className, ...rest } = props;
-    const [scrollTop, setScrollTop] = useState(0);
+    const [scrolled, setScrolled] = useState(false);
+    const [isOverflowing, setIsOverflowing] = useState(false);
 
-    const handleScroll = (event: UIEvent<HTMLElement>) => {
-      setScrollTop(event.currentTarget.scrollTop);
-    };
+    const divRef = useRef<HTMLDivElement>(null);
+    const containerRef = useForkRef(divRef, ref);
+    const handleScroll = debounce(() => {
+      if (!divRef.current) return;
+      setScrolled(divRef.current.scrollTop > 0);
+    }, 50);
```

Comment:
50 is quite fast, was this chosen specifically?

Reply:
i was using the default but it is looks very laggy

Reply:
Have you tried without debouncing? Throttling may be more valid.